### PR TITLE
completed first recurring character problem

### PR DESCRIPTION
**Pull request by goelvatsal**

### CODE REVIEW

File: section7Hash/recurringChar/recurringChar.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package recurringchar
+
+type recurringCharS1 struct{}
+
+func (s1 recurringCharS1) findRecurringChar(input1 []int) int {
+	kvPair := make(map[int]bool, 0)
```
Comment (goel4607) [inline section7Hash/recurringChar/recurringChar.go]: variable names should reflect its purpose like charSet etc

File: section7Hash/recurringChar/recurringChar.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package recurringchar
+
+type recurringCharS1 struct{}
+
+func (s1 recurringCharS1) findRecurringChar(input1 []int) int {
```
Comment (goel4607) [inline section7Hash/recurringChar/recurringChar.go]: specify more meaningful name instead-of input1

File: section7Hash/recurringChar/recurringChar_test.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+type ArrayTester struct {
+	input1         []int
+	firstRecurring int
+}
+
+func getTestData() []ArrayTester {
+	return []ArrayTester{
+		{[]int{2, 5, 1, 2, 3, 5, 1, 2, 4}, 2},
+		{[]int{2, 1, 1, 2, 3, 5, 1, 2, 4}, 1},
+	}
```
Comment (goel4607) [inline section7Hash/recurringChar/recurringChar_test.go:52]: add more test cases tricky ones

File: section7Hash/recurringChar/recurringChar.go (hunk 1 above)
Comment (goelvatsal) [reply section7Hash/recurringChar/recurringChar.go]: changed to charSet

File: section7Hash/recurringChar/recurringChar.go (hunk 2 above)
Comment (goelvatsal) [reply section7Hash/recurringChar/recurringChar.go]: changed to ints

File: section7Hash/recurringChar/recurringChar_test.go (hunk 3 above)
Comment (goelvatsal) [reply section7Hash/recurringChar/recurringChar_test.go:52]: added more test cases

**Review by goel4607 — APPROVED** [review]
Good job

**Pull request merged**